backend/cmd/worker/sample_validation.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	"github.com/golang/glog"
+	"github.com/radiant-network/radiant-api/internal/repository"
+	"github.com/radiant-network/radiant-api/internal/types"
+	"gorm.io/gorm"
+)
+
+const SampleAlreadyExistCode = "SAMPLE-001"
+const SampleExistingSampleDifferentFieldCode = "SAMPLE-002"
+const SampleOrgNotExistCode = "SAMPLE-003"
+const SamplePatientNotExistCode = "SAMPLE-004"
+const SampleUnknownParentSubmitterSampleIdCode = "SAMPLE-005"
+const SampleInvalidValueCode = "SAMPLE-006"
+const SampleInvalidPatientForParentSampleCode = "SAMPLE-007"
+
+type SampleValidationRecord struct {
+	BaseValidationRecord
+	Sample         types.SampleBatch
+	PatientId      int
+	OrganizationId int
+}
+
+func (r SampleValidationRecord) GetBase() *BaseValidationRecord {
+	return &r.BaseValidationRecord
+}
+
+func (r SampleValidationRecord) formatInvalidField(fieldName string, reason string) string {
+	message := fmt.Sprintf("Invalid Field %s for sample (%s / %s). Reason: %s", fieldName, r.Sample.SampleOrganizationCode, r.Sample.SubmitterSampleId, reason)
+	return message
+}
+
+func (r SampleValidationRecord) formatFieldTooLong(fieldName string, maxLength int) string {
```

Comment:
this should be related to BaseValidationRecord ? Because it's not specific to sample